daemon/state.go

```diff
@@ -346,6 +346,14 @@ func (d *Daemon) initRestore(restoredEndpoints *endpointRestoreState) chan struc
 
 		go func() {
 			if k8s.IsEnabled() {
+				// Also wait for all cluster mesh to be synchronized with the
+				// datapath before proceeding.
+				if option.Config.ClusterID != 0 {
+					err := d.clustermesh.ClustersSynced(context.Background())
```

Comment:
Just double-checking, but do we know that at this point all remote clusters are known?

Reply:
We know that at this point all remote clusters are known because they are loaded from the directory. New files can show up after the cilium pod has started which means it's a new cluster and wasn't running before cilium restart.